pydantic/schema.py

```diff
@@ -720,7 +720,7 @@ def field_singleton_schema(  # noqa: C901 (ignore complexity)
             ref_prefix=ref_prefix,
             known_models=known_models,
         )
-    if field.type_ is Any:
+    if field.type_ is Any or type(field.type_) == TypeVar:
```

Comment:
shouldn't be the same as `Any`, should be restricted to a dict, same as if I used

```py
class MyModel(BaseModel):
     foo: dict
```

Reply:
@samuelcolvin this is checking for a TypeVar because the function will recurse down into the typevars on the unparameterized Dict. By putting no restriction, you still get the dict handling, you just aren't adding any constraints on the keys or values.

The upshot of this approach rather than an explicit check for Dict is that it also works for List (or any other unparameterized generic pydantic can handle, though I guess maybe there aren't any besides list where this issue would apply).

I augmented the test to show that this fix does in fact cause the generated schema to be exactly as though you had written
```python
class MyModel(BaseModel):
     foo: dict
```
Let me know if your issue still stands.

Reply:
It looks like bare `Mapping` is also handled properly by this; I suspect that would require another special case to handle if checking for `Dict`. But maybe you have something totally different in mind?

Reply:
No that sounds good, can you add tests for validating unparameterised `Dict` and `List`.

Reply:
> can you add tests for validating unparameterised `Dict` and `List`.

@samuelcolvin I'm not sure what you mean by that -- do you want me to split `test_unparameterized_schema_generation` in two?

There is also already a test of the validation of BaseModel for `List` and `Dict` in `tests/test_types.py` [here](https://github.com/samuelcolvin/pydantic/blob/master/tests/test_types.py#L1649). Would you like me to augment that test to be a little more comprehensive?

Reply:
sorry, I didn't know about those two tests. This is fine.